### DISCUSSION TITLE
Remove PUT routes

### DISCUSSION THREAD

**Discussion by DaTebe** [FraunhoferISST/diva#96] [Q&A] [answered]
We have implemented PUT routes for all entities. However, over time it has become apparent that these routes are not used because their functionality is largely mapped via PATCH. If we do not have the use case "upsert" in DIVA, we can also save the maintenance of the PUT routes. How do you guys see this?

@setaman @anatoliynovo

**Comment by setaman** [comment, marked as answer]
@DaTebe 
yeah, indeed, the PUT routes are used very rarely and currently do net tend to be used more actively. But they are used. Our UM relies on the PUT semantics to handle wired synchronization with Keycloak. The second thing, the PUT is offered to all the services through the `common` package automatically. That means minimal maintaining costs. The main inconvenience is that the endpoint has to be document  in the OpenAPI.

I would say, we can safely remove the PUT route (including documentation) from the services that do not require it explicitly, without to touch the `common` package. But there is no urgency to do or not do that.
